test/integration/toolchains/gnu/test_makedeps.py

```diff
@@ -44,6 +45,7 @@ def package_info(self):
     assert f'CONAN_LIB_DIRS_MYLIB = \\\n\t$(CONAN_LIB_DIR_FLAG){prefix}/my_absoulte_path/fake/mylib/lib \\\n\t$(CONAN_LIB_DIR_FLAG)$(CONAN_ROOT_MYLIB)/lib2' in makefile_content
     assert f'CONAN_INCLUDE_DIRS_MYLIB = $(CONAN_INCLUDE_DIR_FLAG){prefix}/my_absoulte_path/fake/mylib/include' in makefile_content
     assert 'CONAN_BIN_DIRS_MYLIB = $(CONAN_BIN_DIR_FLAG)$(CONAN_ROOT_MYLIB)/bin' in makefile_content
+    assert 'CONAN_PROPERTY_MYLIB_MY_PROP = my prop value' in makefile_content
```

Comment:
How does Make deal with spaces? Should this be quoted?

Reply:
Spaces in paths and `MakeDeps` seems to be a deep rabbit hole. I tested what MakeDeps generates if I make a package editable and have a space in the editable path (even without my changes). I do following in the shell:

```
conan editable add ~/my/path\ with/space 
``` 
MakeDeps wil create a makefile like this when I install the package:
```
CONAN_ROOT_MYLIB = /home/myuser/my/path with/space

CONAN_INCLUDE_DIRS_MYLIB = \
	$(CONAN_INCLUDE_DIR_FLAG)$(CONAN_ROOT_MYLIB)/include \
	$(CONAN_INCLUDE_DIR_FLAG)$(CONAN_ROOT_MYLIB)/h
```
Using `CONAN_INCLUDE_DIRS_MYLIB` in a receipt to pass the include folders to the compiler will not work in this situation. I assume that you don't receive bug reports about this, because no sane developer name their folders like this.

Reply:
Today I found this:

https://savannah.gnu.org/bugs/?712

Because of its weird space handling GNU Make cannot handle paths with spaces correctly, and there seems to be no easy solution for that.

This makes me also prefering not to escape spaces.

Reply:
> Based on my expericence with my own make based build system integration I assume that the variables will land in a receipe (e.g. as compiler/linker options). Therefore I vote on not escaping whitespaces.

Yes, I think I agree, if there is no really better default, the best is not doing or manipulating the inputs at all.

> Because of its weird space handling GNU Make cannot handle paths with spaces correctly, and there seems to be no easy solution for that.

That is true, I recall that we also have a ``TestClient(path_with_spaces=False)`` to allow for tests that won't support in any way paths with spaces, and Make/Autotools tests are common reasons.


Lets not do anything with spaces, and add some comment in the docs that this will be problematic and something on the user side.

Reply:
Ok. During finalization of the docs a last important thought came into my mind. A newline in the property value may break the generated makefile syntax. A makefile with broken syntax is worse than a make variable with wrong value. Therefore, maybe the generoator should convert newlines in spaces.

The alternative would be using the `define`, `endef`syntax if the value contains newlines. However, even that does not prevent the makefile from breaking. As an example:

```
define XXX
two
lines
endef

mytarget:
	echo "$(XXX)"
```

This will not work, because it expands to

```
define XXX
two
lines
endef

mytarget:
	echo "two
lines"
```

But

```
define XXX
echo two
echo lines
endef

mytarget:
	$(XXX)
```

works. And that's why I would convert a multiline value to one line. I cannot imagine any use case that would need multilne properties.

Reply:
There is another approach, it is also possible to:
- Do a warning and do nothing, recommend the users to file a ticket and give feedback (it is more likely that users will realize about the property not being defined than having a modified value)
- Do the same but with a hard error saying it is not supported at the moment

Reply:
I like the policy of "don't break userspace". When using MakeDeps, a hard error would break existing conanfile.py receipts that worked before with this generator. A warning may be a hint for the package publisher but it would not prevent existing custom makefile based build system integrations from breaking if they use packages with receipts that set properties with newlines. There should be no such receipts out there in the wild, but I have the experience, that if something is possible, it is out there somewhere. The only solution to not break any kind of existing integration is to change newlines into spaces or (another new alternative) to give a warning and skip the property that has a newline. So they will not appear in the generated makefile.

Reply:
> I like the policy of "don't break userspace".

I agree that this sounds better for this specific use case.

As the ``CONAN_PROPERTY_`` are not defined yet, not adding one would be perfectly valid, and I think modifying some user property value shouldn't be done, if the value contains multiline, removing them will probably also break, but unexpectedly because the user might use the ``CONAN_PROPERTY_`` that will exist, but with a modified value. 

Then I suggest:
- Do a warning, not an error, so it doesn't break
- Do not generate the ``CONAN_PROPERTY_``

Reply:
Done. Docs updated, too.